internal/database/users.go

```diff
@@ -1204,8 +1205,10 @@ SELECT u.id,
        u.invalidated_sessions_at,
        u.tos_accepted,
        u.searchable,
-       (SELECT COUNT(user_id) FROM user_external_accounts WHERE user_id=u.id AND service_type = 'scim') >= 1 AS scim_controlled
-FROM users u %s`, query)
+       (SELECT COUNT(user_id) FROM scim_accounts WHERE user_id=u.id) >= 1 AS scim_controlled
+FROM users u
+LEFT JOIN scim_accounts sa ON u.id = sa.user_id
```

Comment:
Why can't this just be the following
```suggestion
       COUNT(uea.service_type) >= 1 AS scim_controlled
FROM users u
LEFT JOIN user_external_accounts uea
    ON u.id = uea.user_id
    AND uea.service_type = 'scim'
    AND uea.deleted_at IS NOT NULL
```

Reply:
And then group by

Reply:
But interestingly enough, this actually seems slower. The scim index makes the original one faster

Reply:
Without the index, the other query takes a massive amount of time. Try substituting `'scim'` with `'github'` and see how long it takes. But the one here finishes really fast even without a dedicated index 🤷‍♂️

Reply:
But yeah @eseliger 's solution is the fastest by far, I'd go with that, even when you add the `AND deleted_at IS NULL`